### PR TITLE
[loki] Make /tmp R/W to fix compactor retention

### LINKED ISSUE #609: [loki] /tmp read-only file system
Similar issue to #577 with `/tmp`. I'm getting bombarded with `level=warn ts=2021-08-09T00:03:54.191957114Z caller=marker.go:214 msg="failed to process marks" path=/data/loki/boltdb-shipper-compactor/retention/markers/1628441994354518113 err="open /tmp/marker-view-656536243: read-only file system"`

I've worked around it by adding:
```
extraVolumes:
  - name: temp
    emptyDir: {}
extraVolumeMounts:
  - name: temp
    mountPath: /tmp
```
A+ for allowing extra volume mounts!

### PR DESCRIPTION
**Pull request by almson**
Fixed #609

### CODE REVIEW

**Comment by zanhsieh** [comment]
@almson 
Please sign DCO. Thank you.

**Comment by jortkoopmans** [comment]
@zanhsieh and/or @slim-bean ; could this PR get a review please 😊, now that the DCO has been signed? Thanks @almson for your efforts.

**Review by zanhsieh — APPROVED** [review]

File: charts/loki/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: "v1"
 name: loki
-version: 2.9.0
+version: 2.9.1
```
Comment (zanhsieh) [inline charts/loki/Chart.yaml:3]: now 2.9.2

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/loki/templates/statefulset.yaml [hunk 2]
```diff
@@ -131,6 +129,8 @@ spec:
           persistentVolumeClaim:
             claimName: {{ .Values.persistence.existingClaim }}
   {{- else }}
+        - name: tmp
```
Comment (t106362512) [inline charts/loki/templates/statefulset.yaml:132]: If .Values.persistence.existingClaim is set, the volume tmp will not be created, so maybe this line can be moved to line 109

**Review by t106362512 — CHANGES_REQUESTED** [review]
.

**Review by t106362512 — CHANGES_REQUESTED** [review]
.

**Comment by almson** [comment]
Bumped version and fixed embarrassing oversight.

**Review by zanhsieh — APPROVED** [review]

**Review by torstenwalter — APPROVED** [review]

**Pull request merged**